packages/type-safe-api/scripts/type-safe-api/generators/java/templates/client/auth/apiKeyAuth.ejs

```diff
@@ -0,0 +1,76 @@
+###TSAPI_WRITE_FILE###
+{
+  "dir": "<%- metadata.srcDir %>/auth",
+  "name": "ApiKeyAuth",
+  "ext": ".java",
+  "overwrite": true
+}
+###/TSAPI_WRITE_FILE###<%- include('../../header.partial.ejs', { info }) %>
+
+
+package <%- metadata.packageName %>.auth;
+
+import <%- metadata.packageName %>.ApiException;
+import <%- metadata.packageName %>.Pair;
+
```

Comment:
can the bulk of these Java module templates be read in as a data resource at runtime (possibly allowing for some level of validation with javac)?

Reply:
I think compiling the generated code as part of some kind of integration test would be the way to go here - I've raised https://github.com/aws/aws-pdk/issues/868 to track that :)